### PR TITLE
yarn: add cgroups feature for better dynamic resources management

### LINKED ISSUE #571: yarn: add cgroups feature for better dynamic resources management

### PR DESCRIPTION
**Pull request by mehdibn**
#### Which issue(s) this PR fixes

Fixes #571 

#### Agreements

- [X] I hereby declare this contribution to be licensed under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0).
- [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: tdp_vars_defaults/yarn/yarn.yml [hunk 1]
```diff
@@ -84,6 +90,13 @@ yarn_site:
   yarn.acl.enable: "true"
   yarn.admin.acl: yarn,knox
   yarn.log.server.url: "https://{{ groups['mapred_jhs'][0] | tosit.tdp.access_fqdn(hostvars) }}:{{ mapred_jhs_https_port }}/jobhistory/logs"
+  # To enable cgroups
+  yarn.nodemanager.linux-container-executor.resources-handler.class: "{% if cgroups_enabled %}org.apache.hadoop.yarn.server.nodemanager.util.CgroupsLCEResourcesHandler{% else %}org.apache.hadoop.yarn.server.nodemanager.util.DefaultLCEResourcesHandler{% endif %}"
+  yarn.nodemanager.linux-container-executor.cgroups.mount: "false"
+  yarn.nodemanager.linux-container-executor.cgroups.mount-path: "{{ cgroups_root_dir }}"
+  yarn.nodemanager.linux-container-executor.cgroups.hierarchy: "/yarn"
+  yarn.nodemanager.linux-container-executor.cgroups.strict-resource-usage: "{% if cgroups_enabled %}true{% else %}false{% endif %}"
```
Comment (leopaul36) [inline tdp_vars_defaults/yarn/yarn.yml:98]: Aren't we missing `yarn.nodemanager.container-executor.class: org.apache.hadoop.yarn.server.nodemanager.LinuxContainerExecutor`?

Comment (leopaul36) [reply tdp_vars_defaults/yarn/yarn.yml:98]: Indeed I did think to look my bad.

**Review by rpignolet — COMMENTED** [review]
You do not create the directories with Ansible, you only add them to `tmpfiles.d` but these directories are only created at boot time. It's desired ?

**Comment by mehdibn** [comment]
> You do not create the directories with Ansible, you only add them to `tmpfiles.d` but these directories are only created at boot time. It's desired ?
you're right !

File: roles/yarn/nodemanager/tasks/install.yml [hunk 2]
```diff
@@ -57,3 +57,15 @@
...
+  when: cgroups_enabled
+
+- name: Create yarn cgroups dirs
+  file:
+    path: "{{ item.path }}"
+    state: directory
+    owner: "{{ item.owner }}"
+    group: "{{ item.group }}"
+    mode: "{{ item.mode }}"
+    recurse: yes
```
Comment (rpignolet) [inline roles/yarn/nodemanager/tasks/install.yml:69]: Why do you need `recurse: yes` ?

**Review by rpignolet — APPROVED** [review]

**Pull request merged**